core/src/main/java/io/fabric8/maven/core/util/MavenUtil.java

```diff
@@ -180,10 +184,17 @@ public static boolean hasClass(MavenProject project, String ... classNames) {
     }
 
     /**
-     * Returns the spring boot `application.properties` or an empty properties object if not found
+     * Returns the spring boot configuration (supports `application.properties` and `application.yml`)
+     * or an empty properties object if not found
      */
     public static Properties getSpringBootApplicationProperties(MavenProject project) {
-        return getPropertiesFile(project, "application.properties");
+        URLClassLoader compileClassLoader = getCompileClassLoader(project);
+        URL ymlResource = compileClassLoader.findResource("application.yml");
+        URL propertiesResource = compileClassLoader.findResource("application.properties");
+
+        Properties props = getPropertiesFromYamlResource(ymlResource);
+        props.putAll(getPropertiesResource(propertiesResource));
```

Comment:
So iiur `application.properties` take precedence over `application.yaml`. Is this also the spring-boot behaviour when starting up ?